File: packages/ios-swift/Sources/CinacoinAppKitUI/CinacoinConnectSheet.swift
import SwiftUI
import CinacoinAppKitConfig

/// Cinacoin branded wallet connection sheet
public struct CinacoinConnectSheet: View {
    
    @Environment(\.colorScheme) private var colorScheme
    @Binding var isPresented: Bool
    
    private let onWalletSelect: (WalletConfig) -> Void
    
    public init(
        isPresented: Binding<Bool>,
        onWalletSelect: @escaping (WalletConfig) -> Void = { _ in }
    ) {
        self._isPresented = isPresented
        self.onWalletSelect = onWalletSelect
    }
    
    public var body: some View {
        let theme = colorScheme == .dark ? CinacoinTheme.dark : CinacoinTheme.light
        
        VStack(spacing: 0) {
            // Header
            HStack {
                Text("Connect Wallet")
                    .font(.system(size: 20, weight: .bold))
                    .foregroundColor(theme.text)
                Spacer()
                Button(action: { isPresented = false }) {
                    Image(systemName: "xmark")
                        .font(.system(size: 16, weight: .medium))
                        .foregroundColor(theme.textSecondary)
                }
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 16)
            
            Divider()
                .background(theme.border)
            
            // Wallet List
            ScrollView {
                VStack(spacing: 8) {
                    ForEach(WalletRegistry.recommended, id: \.id) { wallet in
                        WalletRow(
                            wallet: wallet,
                            theme: theme,
                            onSelect: {
                                onWalletSelect(wallet)
                                isPresented = false
                            }
                        )
                    }
                }
                .padding(16)
            }
            
            // Footer
            VStack(spacing: 12) {
                Text("By connecting, you agree to our Terms of Service")
                    .font(.system(size: 12))
                    .foregroundColor(theme.textSecondary)
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 16)
            .background(theme.surface)
        }
        .background(theme.background)
        .clipShape(RoundedRectangle(cornerRadius: 20))
        .presentationDetents([.medium, .large])
    }
}

private struct WalletRow: View {
    let wallet: WalletConfig
    let theme: CinacoinTheme
    let onSelect: () -> Void
    
    var body: some View {
        Button(action: onSelect) {
            HStack(spacing: 12) {
                // Wallet Icon
                AsyncImage(url: URL(string: wallet.iconUrl ?? "")) { image in
                    image
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                } placeholder: {
                    Circle()
                        .fill(theme.surface)
                }
                .frame(width: 40, height: 40)
                .clipShape(Circle())
                
                // Wallet Name
                Text(wallet.name)
                    .font(.system(size: 16, weight: .medium))
                    .foregroundColor(theme.text)
                
                Spacer()
                
                // Chevron
                Image(systemName: "chevron.right")
                    .font(.system(size: 14))
                    .foregroundColor(theme.textSecondary)
            }
            .padding(12)
            .background(theme.surface)
            .clipShape(RoundedRectangle(cornerRadius: 12))
        }
        .buttonStyle(.plain)
    }
}

#Preview {
    CinacoinConnectSheet(isPresented: .constant(true))
}

File: packages/flutter-dart/lib/appkit_ui/cinacoin_connect_sheet.dart
import 'package:flutter/material.dart';
import '../appkit_config/cinacoin_theme.dart';
import '../appkit_config/wallet_registry.dart';

/// Cinacoin branded wallet connection bottom sheet
class CinacoinConnectSheet extends StatelessWidget {
  final void Function(WalletConfig wallet)? onWalletSelect;
  final CinacoinTheme? theme;

  const CinacoinConnectSheet({
    super.key,
    this.onWalletSelect,
    this.theme,
  });

  /// Show the connect sheet as a modal bottom sheet
  static Future<T?> show<T>({
    required BuildContext context,
    void Function(WalletConfig wallet)? onWalletSelect,
    CinacoinTheme? theme,
  }) {
    return showModalBottomSheet<T>(
      context: context,
      backgroundColor: Colors.transparent,
      builder: (context) => CinacoinConnectSheet(
        onWalletSelect: onWalletSelect,
        theme: theme,
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    final themeData = theme ?? _getThemeFromBrightness(context);

    return Container(
      decoration: BoxDecoration(
        color: themeData.background,
        borderRadius: const BorderRadius.vertical(top: Radius.circular(20)),
      ),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          // Header
          Padding(
            padding: const EdgeInsets.all(16),
            child: Row(
              children: [
                Text(
                  'Connect Wallet',
                  style: TextStyle(
                    fontSize: 20,
                    fontWeight: FontWeight.bold,
                    color: themeData.text,
                  ),
                ),
                const Spacer(),
                IconButton(
                  onPressed: () => Navigator.pop(context),
                  icon: Icon(Icons.close, color: themeData.textSecondary),
                ),
              ],
            ),
          ),
          const Divider(),
          
          // Wallet List
          FlexibleSpaceBar.createSettings(
            child: ListView.builder(
              shrinkWrap: true,
              padding: const EdgeInsets.all(16),
              itemCount: WalletRegistry.recommended.length,
              itemBuilder: (context, index) {
                final wallet = WalletRegistry.recommended[index];
                return _WalletRow(
                  wallet: wallet,
                  theme: themeData,
                  onTap: () {
                    onWalletSelect?.call(wallet);
                    Navigator.pop(context);
                  },
                );
              },
            ),
          ),
          
          // Footer
          Container(
            padding: const EdgeInsets.all(16),
            decoration: BoxDecoration(color: themeData.surface),
            child: Text(
              'By connecting, you agree to our Terms of Service',
              style: TextStyle(
                fontSize: 12,
                color: themeData.textSecondary,
              ),
              textAlign: TextAlign.center,
            ),
          ),
        ],
      ),
    );
  }

  CinacoinTheme _getThemeFromBrightness(BuildContext context) {
    final brightness = Theme.of(context).brightness;
    return brightness == Brightness.dark 
        ? CinacoinTheme.dark 
        : CinacoinTheme.light;
  }
}

class _WalletRow extends StatelessWidget {
  final WalletConfig wallet;
  final CinacoinTheme theme;
  final VoidCallback onTap;

  const _WalletRow({
    required this.wallet,
    required this.theme,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: const EdgeInsets.only(bottom: 8),
      decoration: BoxDecoration(
        color: theme.surface,
        borderRadius: BorderRadius.circular(12),
      ),
      child: ListTile(
        onTap: onTap,
        leading: CircleAvatar(
          radius: 20,
          backgroundImage: wallet.iconUrl != null 
              ? NetworkImage(wallet.iconUrl!) 
              : null,
          backgroundColor: theme.surface,
        ),
        title: Text(
          wallet.name,
          style: TextStyle(
            fontSize: 16,
            fontWeight: FontWeight.w500,
            color: theme.text,
          ),
        ),
        trailing: Icon(Icons.chevron_right, color: theme.textSecondary),
      ),
    );
  }
}
